src/cloudant/_common_util.py

```diff
@@ -263,6 +263,16 @@ def get_docs(r_session, url, encoder=None, headers=None, **params):
     resp.raise_for_status()
     return resp
 
+#pylint: disable=unused-argument
+def append_response_error_content(response, **kwargs):
+    """
+    Provides a helper to add HTTP response error and reason messages.
+    """
+    if response.status_code >= 400:
+        resp_content = response.json()
+        response.url += ' %s %s' % (resp_content['error'], resp_content['reason'])
```

Comment:
I think we should add some safety around the extraction of the JSON and `error` and `reason` keys to prevent this method causing a failure if the response isn't JSON (e.g. what if a proxy sends a 40x with some non-JSON message)? Also in case the JSON doesn't contain the keys, IIRC python will raise `KeyError` if you try to access a key that doesn't exist in a dict - and whilst they should always be there we should protect against them being absent.